### PR TITLE
Implement lstat and realpath

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
And fix _Exit false alarm.

### CODE REVIEW

**Comment by yamt** [comment]
symlink as link sounds wrong to me.
an implementation which already returns EPERM would be more correct.

**Comment by xiaoxiang781216** [comment]
> symlink as link sounds wrong to me.
> an implementation which always returns EPERM would be more correct.

From my understanding, the current implement of link is actually symlink:
https://github.com/apache/incubator-nuttx/blob/master/fs/vfs/fs_link.c#L75
link doesn't really implement in NuttX at all.

**Comment by yamt** [comment]
@xiaoxiang781216 ok, thank you for explanation. i didn't know what nuttx's link() is.

btw, what's your motivation to add symlink? just curious.

**Comment by xiaoxiang781216** [comment]
I am upgrading libcxx to the latest version which need symlink to implement C++ 17 filesystem library.

**Comment by yamt** [comment]
@xiaoxiang781216 thank you for info

**Comment by xiaoxiang781216** [comment]
@yamt if you don't any concern, please help me merge the patch.

File: fs/inode/inode.h [hunk 1]
```diff
@@ -248,9 +248,10 @@ int inode_find(FAR struct inode_search_s *desc);
  *   <sys/stat.h>, into which information is placed concerning the file.
  *
  * Input Parameters:
- *   inode - The indoe of interest
- *   buf   - The caller provide location in which to return information about
- *           the inode.
+ *   inode   - The indoe of interest
```
Comment (patacongo) [inline fs/inode/inode.h:251]:
```suggestion
 *   inode   - The inode of interest
```

File: fs/vfs/fs_stat.c [hunk 2]
```diff
@@ -243,17 +257,18 @@ int stat(FAR const char *path, FAR struct stat *buf)
  *   <sys/stat.h>, into which information is placed concerning the file.
  *
  * Input Parameters:
- *   inode - The indoe of interest
- *   buf   - The caller provide location in which to return information about
- *           the inode.
+ *   inode   - The indoe of interest
```
Comment (patacongo) [inline fs/vfs/fs_stat.c:260]:
```suggestion
 *   inode   - The inode of interest
```

File: fs/inode/inode.h (hunk 1 above)
Comment (xiaoxiang781216) [reply fs/inode/inode.h]: Done.

File: fs/vfs/fs_stat.c (hunk 2 above)
Comment (xiaoxiang781216) [reply fs/vfs/fs_stat.c]: Done.

**Comment by xiaoxiang781216** [comment]
> Looks good. I did see two typos that were already in the code. Can we fix those too while we are changing those lines?

Fixed, thanks for review.

**Pull request merged**